hudi-common/src/main/java/org/apache/hudi/common/table/view/AbstractTableFileSystemView.java

```diff
@@ -701,9 +704,9 @@ public final Stream<Pair<String, CompactionOperation>> getPendingLogCompactionOp
   @Override
   public final Stream<HoodieBaseFile> getLatestBaseFiles(String partitionStr) {
     try {
-      readLock.lock();
       String partitionPath = formatPartitionKey(partitionStr);
       ensurePartitionLoadedCorrectly(partitionPath);
+      readLock.lock();
```

Comment:
let's check all the places that the read lock should be acquired outside the scope of the write lock.

Reply:
I've done this and one thing I was considering is to move the read locks closer to the actual access so we can minimize the time holding the lock and reduce the repeated code.

Reply:
Not really, such as 
```java
Map<String, Stream<HoodieBaseFile>> getAllLatestBaseFilesBeforeOrOn(String maxCommitTime)
```
is the missing one.

Reply:
Oops `ensurePartitionLoadedCorrectly(String partition)` has no synchronization on `addedPartitions` while `ensurePartitionsLoadedCorrectly(List<String> partitionList)` does, should we unify the lock: remove the synchronized block in `ensurePartitionsLoadedCorrectly` and add read lock for `ensurePartitionLoadedCorrectly` to ensure the integrity of `addedPartitions` ?

Reply:
I think we should unify the two methods where `ensurePartitionLoadedCorrectly` calls `ensurePartitionsLoadedCorrectly` with a singleton list to reduce repeated code.

Do we really need this extra data structure for tracking `addedPartitions` if there is also a call to `isPartitionAvailableInStore` to check whether the partition is already loaded?